src/_P092_DLbus.ino

```diff
@@ -0,0 +1,1288 @@
+      P092_DataSettings.MWhBytes = 4;
+      P092_DataSettings.IdxCRC = 34;
+
+      P092_OptionValueDecimals[6] = 1; //[0,1kW]     F("Heat power (kW)")
+      break;
+    case 6133:  //UVR 61-3 (from V8.3)
+      P092_DataSettings.DataBytes = 62;
+      P092_DataSettings.DLbus_MinPulseWidth = P092_min_width_488;
+      P092_DataSettings.DLbus_MaxPulseWidth = P092_max_width_488;
+      P092_DataSettings.DLbus_MinDoublePulseWidth = P092_double_min_width_488;
+      P092_DataSettings.DLbus_MaxDoublePulseWidth = P092_double_max_width_488;
+
+      P092_DataSettings.DeviceByte0 = 0x90;
+      P092_DataSettings.DeviceByte1 = 0x9F;
+      P092_DataSettings.DeviceBytes = 2;
+      P092_DataSettings.DontCareBytes = 1;
+      P092_DataSettings.TimeStampBytes = 5;
+      P092_DataSettings.MaxSensors = 6;
+      P092_DataSettings.MaxExtSensors = 9;
+      P092_DataSettings.OutputBytes = 1;
+      P092_DataSettings.SpeedBytes = 1;
+      P092_DataSettings.AnalogBytes = 2;
+      P092_DataSettings.VolumeBytes = 0;
+      P092_DataSettings.MaxHeatMeters = 3;
+      P092_DataSettings.CurrentHmBytes = 2;
+      P092_DataSettings.MWhBytes = 2;
+      P092_DataSettings.IdxCRC = 61;
+
+      P092_OptionValueDecimals[6] = 1; //[0,1kW]     F("Heat power (kW)")
+      break;
+  }
+  P092_DataSettings.IdxSensor               = P092_DataSettings.DeviceBytes + P092_DataSettings.DontCareBytes + P092_DataSettings.TimeStampBytes;
+  P092_DataSettings.IdxExtSensor            = P092_DataSettings.IdxSensor + 2 * P092_DataSettings.MaxSensors;
+  P092_DataSettings.IdxOutput               = P092_DataSettings.IdxExtSensor + 2 * P092_DataSettings.MaxExtSensors;
+  P092_DataSettings.IdxDrehzahl             = P092_DataSettings.IdxOutput + P092_DataSettings.OutputBytes;
+  P092_DataSettings.IdxAnalog               = P092_DataSettings.IdxDrehzahl + P092_DataSettings.SpeedBytes;
+  P092_DataSettings.IdxHmRegister           = P092_DataSettings.IdxAnalog + P092_DataSettings.AnalogBytes;
+  P092_DataSettings.IdxVolume               = P092_DataSettings.IdxHmRegister + 1;
+  P092_DataSettings.IdxHeatMeter1           = P092_DataSettings.IdxVolume + P092_DataSettings.VolumeBytes;
+  P092_DataSettings.IdxkWh1                 = P092_DataSettings.IdxHeatMeter1 + P092_DataSettings.CurrentHmBytes;
+  P092_DataSettings.IdxMWh1                 = P092_DataSettings.IdxkWh1 + 2;
+  P092_DataSettings.IdxHeatMeter2           = P092_DataSettings.IdxMWh1 + P092_DataSettings.MWhBytes;
+  P092_DataSettings.IdxkWh2                 = P092_DataSettings.IdxHeatMeter2 + P092_DataSettings.CurrentHmBytes;
+  P092_DataSettings.IdxMWh2                 = P092_DataSettings.IdxkWh2 + 2;
+  P092_DataSettings.IdxHeatMeter3           = P092_DataSettings.IdxMWh2 + P092_DataSettings.MWhBytes;
+  P092_DataSettings.IdxkWh3                 = P092_DataSettings.IdxHeatMeter3 + P092_DataSettings.CurrentHmBytes;
+  P092_DataSettings.IdxMWh3                 = P092_DataSettings.IdxkWh3 + 2;
```

Comment:
Why not create functions to get the derived values?
In the struct you can create get functions for these.
It will also make sure derived values are always in sync with the source values.
When using it like this, it is possible a source value is updated and the derived value is not, or the other way around.

If the relation to a derived value and source values is always the same, please use a function to get those derived values.

N.B. a get function like that will not increase the size of the struct you define it for.

Reply:
Changed in the next version of _P092_DLbus.ino